src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
 }
 
 
+/*
+ * keeper_ensure_pg_configuration_files_in_pgdata checks if
+ * postgresql.conf, pg_hba.conf, pg_ident.conf files exist
+ * in $PGDATA, if not it tries to get them from default
+ * location and modifies paths inside copied postgresql.conf.
+ */
+static bool
+keeper_ensure_pg_configuration_files_in_pgdata(KeeperConfig *config)
+{
+	char pgdata[MAXPGPATH];
+	char hbaConfPath[MAXPGPATH];
+	char posgresqlConfPath[MAXPGPATH];
+	char *pgdataRelativePath = NULL;
+	char defaultConfigurationPath[MAXPGPATH];
+	bool configurationPathPresent = false;
+	bool postgresqlConfExists = false;
+	char defaultPosgresqlConfPath[MAXPGPATH];
+	char defaultHbaConfPath[MAXPGPATH];
+	char defaultIdentConfPath[MAXPGPATH];
+	char identConfPath[MAXPGPATH];
+
+	get_absolute_path(config->pgSetup.pgdata, pgdata, MAXPGPATH);
+
+	/* check if postgresql.conf exists in pgdata. there is no separate check for pg_hba and pg_ident  */
+	join_path_components(posgresqlConfPath, pgdata, "postgresql.conf");
+	postgresqlConfExists = file_exists(posgresqlConfPath);
+	if (postgresqlConfExists)
+	{
+		return true;
+	}
+
+	/* check if pgdata is in default path. */
+	if (strstr(pgdata, "/var/lib/postgresql") != pgdata)
+	{
+		log_error("Cannot determine configuration file location for this instance. pgdata = %s", pgdata);
```

Comment:
If Postgres is running, we could run a query to discover the configuration file value. We could even run a single query that fetches the file path for the `config_file`, the `hba_file`, and the `pg_ident` file (not sure about that last one) and then fill-in a structure with all the information ready for the rest of this command.